xrootd/encoder/encoder_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package encoder // import "go-hep.org/x/hep/xrootd/encoder"
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go-hep.org/x/hep/xrootd/protocol"
+)
+
+type request struct {
+	Z int64
+	X int32
+	A uint8
+	C uint16
+	D [2]byte
+	Y []byte
```

Comment:
I believe `Marshal` and `Unmarshal` would break if you were to add another field of type `[]byte` here.